src/Core/MailTemplate/FolderThemeScanner.php

```diff
@@ -190,6 +190,16 @@ private function checkThemeFolder($mailThemeFolder)
 
     private function getTemplatePath(SplFileInfo $fileInfo): string
     {
-        return '@MailThemes' . substr($fileInfo->getRealPath(), strlen($this->baseThemeFolder));
+        $realPath = $fileInfo->getRealPath();
+
+        if (_PS_VERSION_ >= '8.0.0') {
+            // If the template comes from a module, we use the @Modules twig path instead of @MailThemes
+            if (str_contains($realPath, _PS_MODULE_DIR_)) {
```

Comment:
I think that you can pass `_PS_MODULE_DIR_` to the constructor

Reply:
What is the benefit of this ?

Reply:
Always better not to depend to global const and it's also easier to unit test as you can provide the path you want

Reply:
Agreed with @atomiix ;)